pkg/serverless/appsec/appsec_test.go

```diff
@@ -32,18 +34,8 @@ func TestNew(t *testing.T) {
 			if stop != nil {
 				defer stop(context.Background())
 			}
-			if err := wafHealth(); err != nil {
-				if ok, _ := waf.SupportsTarget(); ok {
-					// host should be supported by appsec, error is unexpected
-					require.NoError(t, err)
```

Comment:
@RomainMuller: this overall case to "skip" the test when the host platform is not supported sounds valid to me, and the problem rather is the use of `NoError` instead of `Error`.

I think we need to restore this and just change the line to `require.Error(t, err)` when the current host is supported.

Reply:
ok, so we need to skip unsupported GOOS/GOARCH because this is running in the datadog agent CI

Reply:
Updated in https://github.com/DataDog/datadog-agent/pull/37639/commits/deba3ad7e9d566b4d8d95ffed525bfde33e9cd00